balancer/frontend-v3
# feat: capture fatal errors in sentry

# Description

It adds `onError` callbacks using different helpers to capture **fatal sentry errors** for the different types of react queries: 
- handler queries (simulate, build and priceImpact)
- wagmi queries (simulation and execution)

For every captured error event in sentry we keep the original error while adding a top level error name and error message and custom extras according to each scenario: 

**Example for error in Remove liquidity handler simulation:** 

![HandlerQueryError](https://github.com/balancer/frontend-v3/assets/1316240/1c7e4f60-5bab-45a4-bc26-19639fa864ee)
 
...
<img width="1399" alt="additionalData" src="https://github.com/balancer/frontend-v3/assets/1316240/ac739d0f-b912-46fc-84d9-35872db3e1d1">

**Example for execution error in Remove liquidity handler simulation:** 

![WagmiSimulationError](https://github.com/balancer/frontend-v3/assets/1316240/bd6fa3fb-1fec-4a65-85bc-3abd720820f6)

### This PR also:
- Adds user account as sentry user tag (id and username) so that it is faster to locate the user account to impersonate with
- Adds sentry tags for wallet and walletNetwork
-  Adds an initial helper to avoid capturing "Rejected by user" errors (I decided to put in in an isolated file `error-filter.ts` to have a clear place for error filtering).
- Adds a basic component for simulation error alert to the Remove Liquidity form (the UI/UX around this kind of errors should be reviewed later).

<img width="636" alt="simulationError" src="https://github.com/balancer/frontend-v3/assets/1316240/8f373db0-91dd-4380-a210-2b964e655262">

### TODO: 
Once naming and structure is approved I will create another PR to: 

- Move error files to a common `error` directory inside `utils`.
- Add `onError` callback to all the missing flow scenarios (for both handler and wagmi queries).
- Add simulation alert to 

## Type of change


## How should this be tested?

Go to remove liquidity for this bricked pool that is currently failing in the SDK simulation: 
http://localhost:3000/pools/ethereum/v2/0xae8535c23afedda9304b03c68a3563b75fc8f92b0000000000000000000005a0/remove-liquidity

Impersonate with a holder: 
`0x489A3AA83C1F204f0647C67C9FBF3e7Ee1463Bc5`

You should see this: 

<img width="636" alt="simulationError" src="https://github.com/balancer/frontend-v3/assets/1316240/ca8f43c0-5e97-4a77-8d99-97cd383cb185">

with an error in the console: 
<img width="1898" alt="simulationErrorConsole" src="https://github.com/balancer/frontend-v3/assets/1316240/dcf23e5c-25f3-4927-9acb-96b15af85848">

You should always have a new error like this one in Sentry:

<img width="942" alt="Screenshot 2024-03-19 at 18 37 57" src="https://github.com/balancer/frontend-v3/assets/1316240/1a9c8239-5f8a-45d4-bab4-2b1b8a724fa5">


## Checklist:
